### DISCUSSION TITLE
Paths that have steps?

### DISCUSSION THREAD

**Discussion by timothywashere** [streetcomplete/StreetComplete#3088] [Ideas]
Hello Again,

I've searched but can't find anything relevant in other discussions

Sometimes a path in Street Complete should actually be steps, or part of it should.
It's obviously sometimes difficult to tell the difference between a path with or without steps from the satellite imagery and so a lot of mapped paths might appear to be accesible on OSM when they aren't.

Could there be some way of turning a path into steps, perhaps related to the Surface type question, allowing you to split the way. Or a new quest, "Does this path have steps?"

Sorry if it's been asked before, Thanks for taking the time to read this :)

**Comment by mnalis** [comment]
There is always `Other answers` / `Can't say` / `Leave a note instead`, where you can explain the situation (perhaps even with a picture) so it can be solved later (or by others). (There is also **long press** + `open location in another app` + `Vespucci` which does provide one with full-fledged editor for any cases where such power is required).

It did happen to me however, but very rarely (maybe several times a year?) - likely not often enough to warrant separate quests/answers. Does it happen very often in your case @timothywashere ?

**Reply by timothywashere** [reply]
Yeah, all the time 
I live in a hilly area.
It's usually paths I've added myself since there aren't many active mappers in my area🤨
I some times leave a note but I prefer to keep walking as I've got a lot of ground to cover.
Would need nice if it was an optional, low priority quest, for when an area has few others. If it just set the fixme flag it would probably be easy to fix using the current high quality BING imagery. A lot of paths where added using the lower quality ESRI imagery, when that was the best available and you can't really see steps Vs paths

**Reply by andrewharvey** [reply]
I agree it's quite common when someone is mapping footways from imagery, they usually can't tell if there are steps or not, so this is well suited for someone doing StreetComplete to pickup on. And knowing if there are steps or not is incredibly important for pedestrian routing.

I like the idea of tying it in as another answer into the surface quest as surface usually also won't be set precisely from imagery.

A separate quest has the problem that you can't really tag path is correct, there are no steps.

**Reply by timothywashere** [reply]
Well, there is "steps=no" which is used in some places, but very rarely

It's also possible it could be part of a Wheelchair quest. 
Just because a footpath doesn't have steps, doesn't mean it's accessible.
Is this path wheelchair accessible? Yes / No - it's too steep / No - it has steps / No - it's too narrow / No - it's too uneven / No - there is a small gate  etc.

**Comment by matkoniecz** [comment]
In such cases I 

(1) split the way

(2) create a note for part with steps "this section is steps not footway"

(3a) (optional) edit it directly with Vespucci and close the note

(3b)  (optional) edit with JOSM or iD at home later


> Or a new quest, "Does this path have steps?"

Marking every single footway "actually not a steps" is not a good idea.

**Reply by andrewharvey** [reply]
> (2) create a note for part with steps "this section is steps not footway"

If I can split the way in streetcomplete already, would it be an issue to have the alternative answer "actually steps here" just go ahead and retag the footway as steps?

**Reply by matkoniecz** [reply]
Technically it is possible - the question is whether it is worth extra interface complexity and translating effort?

Implementation would be fairly easy, just an extra answer that changes `highway` value.

If someone feels that it would be a good addition and it was not done before then creating an issue would be the next step.

**Reply by peternewman** [reply]
Isn't the main benefit that you can then go on to answer the remaining step quests about the new steps immediately and accurately (and without having to write more notes)?

In the same way as you can go e.g. diet related quest->shop isn't here anymore->different type of food place, name of shop, diet related quest on new shop etc

**Reply by mnalis** [reply]
Note that also sometimes there is separate steps going alongside a footway uphill, so it should be pointed to the user if it is going to be implemented this way (in order not to damage/update the wrong way - kinda like we have `sidewalk`/`cycleway` answer `yes, displayed separately on the map`)

**Reply by peternewman** [reply]
Having hit this issue again, I've opened #3288

**Comment by timothywashere** [comment]
Related question: When does a footpath become steps?
If a footpath has a single step half way along, should it be marked as three seperate connected ways, one being "Steps" with a step count of 1?

**Reply by matkoniecz** [reply]
>  If a footpath has a single step half way along, should it be marked as three seperate connected ways, one being "Steps" with a step count of 1?

I map this way, though some people put `barrier=step` node inside the way
